### PR TITLE
Cleanup Code

### PR DESCRIPTION
**Pull request by 6543**
* Fix "Empty slice declaration using a literal"
* Fix "collides with imported package name"
* Remove unused code in pipeline
* Remove unused oauth2.providerAuthHeaderWorks()
* Add TODOs
* Format Code
* Cleanup doublestar import
* Migrate deprecated functions

### CODE REVIEW

File: server/store/datastore/init.go [hunk 1]
```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//go:build !cgo
```
Comment (6543) [inline server/store/datastore/init.go:15]: INFO: this is used by newer go versions - we keep the old one too to be backwards compatible

File: server/shared/configFetcher.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ func NewConfigFetcher(remote remote.Remote, user *model.User, repo *model.Repo,
 
 func (cf *configFetcher) Fetch() (files []*remote.FileMeta, err error) {
 	var file []byte
+	cf.repo.Config = strings.TrimSpace(cf.repo.Config)
```
Comment (anbraten) [inline server/shared/configFetcher.go:29]: This should be done at in post func saving the string to the database.

**Review by anbraten — COMMENTED** [review]
Could we introduce some kind of linter like golangci to automatically check most of your changes.

File: pipeline/frontend/yaml/constraint.go [hunk 3]
```diff
@@ -5,7 +5,7 @@ import (
 	"path/filepath"
 	"strings"
 
-	doublestar "github.com/bmatcuk/doublestar/v4"
+	"github.com/bmatcuk/doublestar/v4"
```
[the remark this answers is not in the document]
Comment (6543) [reply pipeline/frontend/yaml/constraint.go:8]: yes go do understand versions

File: server/shared/configFetcher.go (hunk 2 above)
Comment (6543) [reply server/shared/configFetcher.go:29]: oh sorry did add int unintentional should be in a extra pull :) - will rm it

**Review by anbraten — APPROVED** [review]

**Pull request merged**